### PR TITLE
Creating Users Table Using Sequelize

### PR DESCRIPTION
**Pull request by MohannadSabaa**

### CODE REVIEW

File: src/db/connection.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Sequelize } from 'sequelize';
+import { PASS, NAME, DATABASE } from '../config';
+
+const sequelize = new Sequelize({
```
Comment (LinaYahya) [inline src/db/connection.ts:4]: as we're going to have a database URL for production, So I'd prefer to not specify every single attribute of db connection, Instead, we can provide `Sequelize` with database connection URL as follows,

Comment (LinaYahya) [reply src/db/connection.ts:4]: please set logging to `false` as it's a bit annoying to log every db operation

File: src/db/connection.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  username: NAME,
+  password: PASS,
+  database: DATABASE,
+  host: 'localhost',
+  port: 5432,
+  dialect: 'postgres',
```
Comment (LinaYahya) [inline src/db/connection.ts:10]: add `ssl`

File: src/db/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { DataTypes } from 'sequelize';
+import UsersAttributes from '../types/usersAttributes';
+import sequelize from './connection';
+
+const Users = sequelize.define<UsersAttributes>('users', {
```
Comment (LinaYahya) [inline src/db/index.ts:5]: model name is preferred to be singular, please check [Sequelize singular-vs-plural](https://sequelize.org/docs/v6/other-topics/naming-strategies/#singular-vs-plural)
```suggestion
const User = sequelize.define<UsersAttributes>('user', {
```

File: src/db/index.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  email: {
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: true,
+  },
+  password: {
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: false,
```
Comment (LinaYahya) [inline src/db/index.ts:23]: the default is false, SO no need to specify the value
```suggestion
```

File: src/db/index.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  },
+  phone_number: {
+    type: DataTypes.STRING,
+  },
+  role: {
+    type: DataTypes.STRING,
+    defaultValue: 'client',
+  },
+  is_active: {
```
Comment (LinaYahya) [inline src/db/index.ts:32]:
```suggestion
  isActive: {
```

File: src/db/index.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: false,
+  },
+  phone_number: {
+    type: DataTypes.STRING,
+  },
+  role: {
+    type: DataTypes.STRING,
+    defaultValue: 'client',
```
Comment (LinaYahya) [inline src/db/index.ts:30]: please check `enum` datatype
[Enum](https://sequelize.org/docs/v6/other-topics/other-data-types/#enums)

File: src/db/index.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: true,
+  },
+  password: {
+    type: DataTypes.STRING,
+    allowNull: false,
+    unique: false,
+  },
+  phone_number: {
```
Comment (LinaYahya) [inline src/db/index.ts:25]:
```suggestion
  phoneNumber: {
```

File: src/db/index.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  role: {
+    type: DataTypes.STRING,
+    defaultValue: 'client',
+  },
+  is_active: {
+    type: DataTypes.BOOLEAN,
+    defaultValue: true,
+  },
+});
+(async () => Users.sync())();
```
Comment (LinaYahya) [inline src/db/index.ts:37]:
```suggestion
```

File: src/types/usersAttributes.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+interface UsersAttributes extends Model
+<InferAttributes<UsersAttributes>,
+InferCreationAttributes<UsersAttributes>> {
+  id: CreationOptional<number>,
+  fullName: string,
+  email: string,
+  password: string,
+  phone_number?: string | undefined,
+  role: CreationOptional<string>,
+  is_active: CreationOptional<boolean>,
```
Comment (LinaYahya) [inline src/types/usersAttributes.ts:14]: When creating a new user, sequelize will automatically add `createdAt`, and `updatedAt` fields, for those we're going to need to add them to type 
```suggestion
  is_active: CreationOptional<boolean>,
  createdAt?: CreationOptional<Date>;
  updatedAt?: CreationOptional<Date>;
```

File: src/types/usersAttributes.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import {
```
Comment (LinaYahya) [inline src/types/usersAttributes.ts:1]: I'd prefer to name file as models, to have all model types after that at this file

File: src/db/index.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import { DataTypes } from 'sequelize';
+import UsersAttributes from '../types/usersAttributes';
```
Comment (LinaYahya) [inline src/db/index.ts:2]: please import it from `/types` index file

File: src/config/index.ts [hunk 12]
```diff
@@ -2,7 +2,12 @@ import dotenv from 'dotenv';
 
 dotenv.config();
 
-const { PORT } = process.env;
+const {
+  PORT, NAME,
+  PASS, DATABASE,
+} = process.env;
```
Comment (LinaYahya) [inline src/config/index.ts:8]: create a config object that have all our configuration
```js
const CONFIG = {
DB : {
DEVELOPMENT: '',
PRODUCTION: ''
},
PORT,
}

and export this object

**Review by LinaYahya — CHANGES_REQUESTED** [review]
Good Job @MohannadSabaa , a few suggestions to enhance our work

File: src/db/connection.ts (hunk 2 above)
Comment (MohannadSabaa) [reply src/db/connection.ts:10]: ssl not supported !!!

File: src/db/index.ts (hunk 4 above)
Comment (MohannadSabaa) [reply src/db/index.ts:23]: done.

File: src/db/index.ts (hunk 8 above)
Comment (MohannadSabaa) [reply src/db/index.ts:37]: done

File: src/types/usersAttributes.ts (hunk 9 above)
Comment (MohannadSabaa) [reply src/types/usersAttributes.ts:14]: done

File: src/db/index.ts (hunk 5 above)
Comment (MohannadSabaa) [reply src/db/index.ts:32]: done

File: src/types/usersAttributes.ts (hunk 10 above)
Comment (MohannadSabaa) [reply src/types/usersAttributes.ts:1]: done

File: src/db/index.ts (hunk 11 above)
Comment (MohannadSabaa) [reply src/db/index.ts:2]: done

File: src/db/index.ts (hunk 3 above)
Comment (MohannadSabaa) [reply src/db/index.ts:5]: done

**Review by raghdarizq — APPROVED** [review]

**Review by ahmedmurtaja — APPROVED** [review]

**Review by YasserAmro0 — APPROVED** [review]

**Review by raghdarizq — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**